src/QUMainWindow.cpp

```diff
@@ -993,9 +994,10 @@ void QUMainWindow::aboutMediaInfo() {
 }
 
 void QUMainWindow::checkForUpdate(bool silent) {
-	int currentVersion = MAJOR_VERSION*100 + MINOR_VERSION*10 + PATCH_VERSION;
+	QString currentVersion = QString(xstr(VERSION));
+	int currentVersionNum = QString(currentVersion).remove('.').toInt();
 
-	QUrl url("https://github.com/UltraStar-Deluxe/UltraStar-Manager/blob/master/src/latest_version.xml");
+	QUrl url("https://api.github.com/repos/UltraStar-Deluxe/UltraStar-Manager/releases/latest");
```

Comment:
Changed the update check to read the latest release tag, rather than this file so we don't have to set it manually. Theoretically, this will break the update check for old versions, but it already just didn't work before as far as I could tell, since the `QNetworkReply` object below contained an entire website in HTML, rather than just the number.

On a side note, this still doesn't work on Windows ("Host not Reachable") and I don't know why.

Reply:
Breaking the update functionality for old versions is not critical, I think. I'm pretty sure it worked at some point, but maybe the Github side was changed at some point, who knows. Anyways, I guess we should try to fix the "Host not reachable" issue to make it work. I will have a look at it probably tomorrow.